core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -380,13 +386,19 @@ private BasicInvocationStage handleNonTxWriteCommand(InvocationContext ctx, Writ
          }
          if (!(ce instanceof OutdatedTopologyException) && !(ce instanceof SuspectException))
             throw t;
+         WriteCommand writeCommand = (WriteCommand) rCommand;
+         if (writeCommand.hasFlag(Flag.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            // the command is expected to not wait
```

Comment:
Actually, the command is expected not to wait for another operation modifying the same lock, the flag javadoc doesn't say anything about other kinds of waiting.